src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,165 @@
+    private static final String BYE_COMMAND = "bye";
+    private static final String DONE_COMMAND = "done";
+    private static final String DELETE_COMMAND = "delete";
+    private static final String TODO_COMMAND = "todo";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String EVENT_COMMAND = "event";
+    private static final String FIND_COMMAND = "find";
+    private static final String TODO_ERROR_MESSAGE = "Oops! The description of todo cannot be empty.";
+    private static final String DEADLINE_ERROR_MESSAGE = "Oops! The description of deadline cannot be empty.";
+    private static final String EVENT_ERROR_MESSAGE = "Oops! The description of event cannot be empty.";
+    private static final String INVALID_COMMAND_MESSAGE = " Oops! Im sorry, but I don't know what that means :(";
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for Duke
+     *
+     * @param filePath path of the file where tasks will be stored.
+     */
+    public Duke (String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public static void main (String[] args) {
+        new Duke("data/duke.txt").run();
+    }
+
+    /**
+     * Executes Duke.
+     */
+    public void run() {
+        ui.greetUser();
+        Scanner scanner =  new Scanner(System.in);
+        boolean isExit = false ;
+        while (!isExit) {
+            try {
+                String input = scanner.nextLine().trim();
+                String[] parsedInput = Parser.parseInput(input);
+                int inputLength = parsedInput.length;
+                List<String> parsedList = Arrays.asList(parsedInput);
+                String command = parsedInput[0];
+                List<String> arguments = parsedList.subList(1, parsedList.size());
+
+                switch (command) {
+                case BYE_COMMAND:
+                    storage.writeNewData(tasks);
+                    ui.exitProgram();
+                    isExit = true;
+                    break;
+
+                case LIST_COMMAND:
+                    ui.list(tasks);
+                    break;
+
+                case DONE_COMMAND:
+                    if (inputLength == 2 && Parser.isNumeric(parsedInput[1])) {
+                        int taskNumber = Integer.parseInt(parsedInput[1]);
+                        try {
+                            tasks.markAsDone(taskNumber - 1);
+                            ui.showDoneMessage(tasks.getTasks().get(taskNumber-1).toString());
+                        } catch (IndexOutOfBoundsException e) {
+                            ui.showTaskErrorMessage();
+                        }
+                    }
+                    else {
```

Comment:
should there be a line break before `else`?

Reply:
Oh i just saw the coding standard...there shouldn't be a line break, thanks prof!